### PR TITLE
Refactor folding_ranges onto POIs, Add support for records.

### PR DESCRIPTION
**Pull request by tks2103**
Moving folding_range from functions to a general field on each poi. Added support for records. Refactored support for functions. Added foldingrange_provider test

### CODE REVIEW

**Review by robertoaloi — APPROVED** [review]

**Comment by robertoaloi** [comment]
Thanks a lot for this @tks2103 ! Great generalization!

CI is failing, but I believe it's because many tests depend on the content of the `code_navigation` test module. We could either adjust the expectations or have a specific test file for folding ranges and point to that in the test suite.

File: apps/els_lsp/priv/code_navigation/src/folding_ranges.erl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+-module(code_navigation).
```
Comment (tks2103) [inline apps/els_lsp/priv/code_navigation/src/folding_ranges.erl:1]: oops, sec

**Comment by tks2103** [comment]
@robertoaloi is this passing CI now? Can't figure out if it ran another set of tests...

I don't have commit access, feel free to merge if it's good.

**Comment by robertoaloi** [comment]
> @robertoaloi is this passing CI now? Can't figure out if it ran another set of tests...

Since this is your first contribution to the project, GitHub requires me to manually approve the CI run. Triggered it now, so let's see. You can also run this manually on your machine, following the steps at:

https://github.com/erlang-ls/erlang_ls/blob/main/.github/workflows/build.yml

So, essentially, you can:

    rebar3 do compile, ct, lint, proper, dialyzer, xref, edoc

**Comment by robertoaloi** [comment]
Thanks!

**Comment by robertoaloi** [comment]
And credit should also go to @AminArria who provided the original implementation in #1266.

**Pull request merged**